homeassistant/components/hvv_departures/config_flow.py

```diff
@@ -0,0 +1,249 @@
+    {
+        vol.Required(CONF_HOST, default=GTI_DEFAULT_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+SCHEMA_STEP_STATION = vol.Schema({vol.Required(CONF_STATION): str})
+
+SCHEMA_STEP_OPTIONS = vol.Schema(
+    {
+        vol.Required(CONF_FILTER): vol.In([]),
+        vol.Required(CONF_OFFSET, default=0): vol.All(int, vol.Range(min=0)),
+        vol.Optional(CONF_REAL_TIME, default=True): bool,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for HVV."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize component."""
+        self.hub = None
+        self.data = None
+        self.stations = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=SCHEMA_STEP_USER)
+
+        errors = {}
+        session = aiohttp_client.async_get_clientsession(self.hass)
+        self.hub = GTIHub(
+            user_input[CONF_HOST],
+            user_input[CONF_USERNAME],
+            user_input[CONF_PASSWORD],
+            session,
+        )
+
+        try:
+            response = await self.hub.authenticate()
+            _LOGGER.debug("init gti: %r", response)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+
+        if not errors:
+            self.data = user_input
+
+            return self.async_show_form(
+                step_id="station", data_schema=SCHEMA_STEP_STATION, errors=errors
+            )
+
+        return self.async_show_form(
+            step_id="user", data_schema=SCHEMA_STEP_USER, errors=errors
+        )
+
+    async def async_step_station(self, user_input=None):
+        """Handle the step where the user inputs his/her station."""
+        if user_input is None:
+
+            return self.async_show_form(
+                step_id="station", data_schema=SCHEMA_STEP_STATION
+            )
+
+        errors = {}
+
+        check_name = await self.hub.gti.checkName(
+            {"theName": {"name": user_input[CONF_STATION]}, "maxList": 20}
+        )
+
+        stations = check_name.get("results")
+
+        if not stations:
+            errors["base"] = "no_results"
+
+            return self.async_show_form(
+                step_id="station", data_schema=SCHEMA_STEP_STATION, errors=errors
+            )
+
+        self.stations = {
+            f"{station.get('name')} ({station.get('type')})": station
+            for station in stations
+        }
+
+        schema = vol.Schema(
+            {vol.Required(CONF_STATION): vol.In(list(self.stations.keys()))}
+        )
+
+        return self.async_show_form(
+            step_id="station_select", data_schema=schema, errors=errors
```

Comment:
Please don't move to another step's form directly. The form and its errors are private to each step. Return and await the next step method instead.